packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3151,7 +3151,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       _textInputConnection = _needsAutofill && currentAutofillScope != null
         ? currentAutofillScope!.attach(this, _effectiveAutofillClient.textInputConfiguration)
         : TextInput.attach(this, _effectiveAutofillClient.textInputConfiguration);
-      _updateSizeAndTransform();
+      // It has to be run in post-frame callback, because when the current
+      // function is executed, we may be in build phase, while the
+      // _updateSizeAndTransform may need layout phase information.
+      // See https://github.com/flutter/flutter/issues/126312 for more details.
+      if (canTouchLayoutPhaseData) {
```

Comment:
I was thinking it would be nice to be able to detect when the error would happen here, such as with `hasSize`, rather than relying on canTouchLayoutPhaseData.  But I guess it's the FractionalTranslation's RenderBox where `hasSize` is false, not the RenderEditable...

Reply:
> But I guess it's the FractionalTranslation's RenderBox where hasSize is false, not the RenderEditable...

Yes...

Indeed, `renderEditable.size` is called one line above `renderEditable.getTransrormTo()` and the former call is happy, so RenderEditable must be laid out.

Reply:
*However*, IMHO the data (renderEditable.size) is incorrect even if it is laid out. It is the size in *last* frame, while in current frame it could be moved a lot!

Reply:
Good point.  Maybe that's another argument that the _openInputConnection call should be in the post frame callback, like you've changed it to.